pkg/security/controller/namespace_scc_allocation_controller.go

```diff
@@ -0,0 +1,339 @@
+
+	actualRangeAllocation, err := c.rangeAllocationClient.RangeAllocations().Update(newRangeAllocation)
+	if err != nil {
+		return err
+	}
+	c.currentUIDRangeAllocation = actualRangeAllocation
+
+	block, ok := uidRange.BlockAt(uint32(bitIndex))
+	if !ok {
+		return fmt.Errorf("%d not in range", bitIndex)
+	}
+
+	// Now modify the namespace
+	nsCopy := ns.DeepCopy()
+	if nsCopy.Annotations == nil {
+		nsCopy.Annotations = make(map[string]string)
+	}
+	nsCopy.Annotations[security.UIDRangeAnnotation] = block.String()
+	nsCopy.Annotations[security.SupplementalGroupsAnnotation] = block.String()
+	if _, ok := nsCopy.Annotations[security.MCSAnnotation]; !ok {
+		if label := c.mcsAllocator(block); label != nil {
+			nsCopy.Annotations[security.MCSAnnotation] = label.String()
+		}
+	}
+
+	_, err = c.namespaceClient.Update(nsCopy)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	success = true
+	return nil
+}
+
+// allocateNextContiguousBit finds a free bit in the int and returns which one it is and whether it succeeded
+func allocateNextContiguousBit(allocated *big.Int, max int) (int, bool) {
+	for i := 0; i < max; i++ {
+		if allocated.Bit(i) == 0 {
+			return i, true
+		}
+	}
+	return 0, false
+}
+
+func (c *NamespaceSCCAllocationController) WaitForRepair(stopCh <-chan struct{}) error {
+	return wait.PollImmediate(10*time.Second, 5*time.Minute, func() (bool, error) {
+		select {
+		case <-stopCh:
+			return true, nil
+		default:
+		}
+		err := c.Repair()
+		if err == nil {
+			return true, nil
+		}
+		utilruntime.HandleError(err)
+		return false, nil
+	})
+}
+
+func (c *NamespaceSCCAllocationController) Repair() error {
+	// TODO: (per smarterclayton) if Get() or List() is a weak consistency read,
+	// or if they are executed against different leaders,
+	// the ordering guarantee required to ensure no item is allocated twice is violated.
+	// List must return a ResourceVersion higher than the etcd index Get,
+	// and the release code must not release items that have allocated but not yet been created
+	// See #8295
+
+	// get the curr so we have a resourceVersion to pin to
+	uidRange, err := c.rangeAllocationClient.RangeAllocations().Get(rangeName, metav1.GetOptions{})
+	needCreate := apierrors.IsNotFound(err)
+	if err != nil && !needCreate {
+		return err
+	}
+	if needCreate {
+		uidRange = &securityv1.RangeAllocation{ObjectMeta: metav1.ObjectMeta{Name: rangeName}}
+	}
+
+	uids := uidallocator.NewInMemory(c.requiredUIDRange)
+	nsList, err := c.nsLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	for _, ns := range nsList {
+		value, ok := ns.Annotations[security.UIDRangeAnnotation]
+		if !ok {
+			continue
+		}
+		block, err := uid.ParseBlock(value)
+		if err != nil {
+			continue
+		}
+
+		switch err := uids.Allocate(block); err {
+		case nil:
+		case uidallocator.ErrNotInRange, uidallocator.ErrAllocated:
+			continue
+		case uidallocator.ErrFull:
+			// TODO: send event
```

Comment:
yeah, this requires user action but how will user know that? ;-)

Reply:
> yeah, this requires user action but how will user know that? ;-)

This is pre-existing.  Same as before.  cluster-admin has log access.